contracts/secret-storage/SecretManager.sol

```diff
@@ -517,26 +549,43 @@ contract SecretManager is
         emit SecretEndTimestampUpdated(_secretId, _endTimestamp);
     }
 
+    /**
+     * @dev Checks if the secret has been acknowledged by the first set of selected enclaves.
+     *      Replacement store selected post slashing of the previous store, isn't taken into account.
+     */
     function _checkIfSecretAcknowledged(
         uint256 _secretId
-    ) internal view {
+    ) internal view returns (bool) {
         uint256 len = userStorage[_secretId].selectedEnclaves.length;
         for (uint256 index = 0; index < len; index++) {
-            if(!userStorage[_secretId].selectedEnclaves[index].hasAcknowledgedStore)
-                revert SecretManagerUnacknowledged();
+            if(!_isReplacedStore(_secretId, index) && !userStorage[_secretId].selectedEnclaves[index].hasAcknowledgedStore)
+                return false;
         }
+        return true;
     }
 
     function _terminateSecret(
         uint256 _secretId
     ) internal {
```

Comment:
lets restrict this method to be called only before the termination time. Otherwise, it will usage will conflict with `_removeSecret`. Keep the flow independent.